source/core_json.c

```diff
@@ -36,10 +36,19 @@ typedef enum
     false = 0
 } bool_;
 
-#define isdigit_( x )    ( ( x >= '0' ) && ( x <= '9' ) )
-#define iscntrl_( x )    ( ( x >= '\0' ) && ( x < ' ' ) )
+/* A compromise to satisfy both MISRA and CBMC */
+typedef union
```

Comment:
Doesn't MISRA dislike unions even more though (Rule 19.2)? That's only an advisory though

Reply:
Yes, I'm choosing to not comply with the advisory against unions, because all the other solutions are worse (either running afoul of MISRA required rules, or throwing up sign conversion errors in CBMC).